src/opdcodes.rs

```
@@ -160,7 +162,7 @@ pub fn trap(vm: &mut VM, instruction: u16) -> Result<(), VMError> {
             vm.state = VMState::Halted;
             Ok(())
         }
-        _ => std::process::exit(1),
```

I think you should return an error here.

---

Done